# HG changeset patch
# User Rik <rik@octave.org>
# Date 1400253611 25200
#      Fri May 16 08:20:11 2014 -0700
# Node ID 10082673a8f0739997a4b342843969e3978e4af7
# Parent  692e16c9e4888685d035d14d9ed0225f3cc48aad
Don't delete listeners with 'cla reset' for Matlab compatibility (bug #42285).

* graphics.in.h (set_defaults): Delete call to remove_all_listeners().

diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -4153,17 +4153,16 @@ public:
       // default_properties map.
       default_properties.set (name.substr (7), value);
     else
       xproperties.set (name, value);
   }
 
   void set_defaults (const std::string& mode)
   {
-    remove_all_listeners ();
     xproperties.set_defaults (*this, mode);
   }
 
   octave_value get (const caseless_str& name) const
   {
     octave_value retval;
 
     // FIXME: finish this.
